sdk/nodejs/output.ts

```diff
@@ -114,6 +114,52 @@ export class Output<T> {
         return obj && obj.__pulumiOutput ? true : false;
     }
 
+    /**
+     * [toString] on an [Output<T>] is not supported.  This is because the value an [Output] points
+     * to is asynchronously computed (and thus, this is akin to calling [toString] on a [Promise]).
+     * As such, this method only throws if called.
+     *
+     * To get the value of an Output<T> as an Output<string> consider either:
+     * 1. `o.apply(v => ``prefix${v}suffix``)` or
+     * 2. `pulumi.interpolate ``prefix${v}suffix`` `
+     *
+     * This will return an Output with the inner computed value and all resources still tracked.
+     */
+    public toString(): string {
+        // Note: i've mixed/matched `` and "" strings because I could not find a way to
+        // properly embed literals with `` inside another `` string.
+        const errorMessage =
+`${"Calling [toString] on an [Output<T>] is not supported."}
```

Comment:
This is a little mindbending - it would be a lot more clear to me what's going on here if we either escaped the `s if that's possible or explicitly concatenating strings with newlines.

Reply:
I'm pretty ok with how this is :)

Reply:
cleaned up.